src/webview/utils/vscode-contexts.ts

```diff
@@ -28,7 +28,24 @@ export interface VscodeContext {
 export type WebviewSection = 'optionsWidget' | 'advancedOptionsOverlay' | 'memoryTable';
 
 export function createVscodeContext<C extends {}>(context: C): VscodeContext {
-    return { 'data-vscode-context': JSON.stringify(context) };
+    return { 'data-vscode-context': JSON.stringify(includeFlatKeys(context)) };
+}
+
+function includeFlatKeys(src: object): Record<string, unknown> {
+    return { ...src, ...flattenKeys(src) };
+}
+
+// VSCode context cannot make use of nested keys in 'when' clauses.
+function flattenKeys(src: object, dst: Record<string, unknown> = {}, prefix = 'memory-inspector.'): Record<string, unknown> {
+    if (!src || typeof src !== 'object') { return dst; }
+    for (const [key, value] of Object.entries(src)) {
+        if (value && typeof value === 'object' && !Array.isArray(value)) {
+            flattenKeys(value, dst, `${prefix}${key}.`);
```

Comment:
Might be a matter of taste. But I prefer the way you implemented it here (with the dots) over a top-level key as you proposed as an alternative in the description. This makes the key easier to read and helps to understand the hierarchy of the key.